06-code.Rmd

```diff
@@ -0,0 +1,223 @@
+
+_This is the beauty of reproducibility, if you or someone else makes great code, it can not only be re-run but it can be repurposed!_
+
+### Tips for choosing packages to use:
+
+- Does the package have easy to use documentation to help guide you on how to use it properly? If the package has underdeveloped documentation it may be difficult for you or other to understand and use properly.
+- Is the code actively being developed or maintained? Packages that are no longer being maintained will likely deprecate more quickly rendering them unusable in the future.
+- Is it a package that is commonly recognized by the community? Well recognized packages will be easier for others to comment on and help you with.
+
+## The importance of iterative work
+
+We've mentioned previously that reproducibility is iterative work. This way of working refers to code work as much as anything else. You won't ever right perfect code on the first try, instead aim for each chunk to work one step at a time. Once it is working, take a break (perhaps until the next day), then return to it and look for ways to polish it and make it more efficient.
+
+For [more tips on how to work read this blog](https://www.ccdatalab.org/blog/not-so-secret-sauce-for-efficient-workflows).
+
+## Aspects of Reproducible code
+
+```{r, fig.align='center', out.width="100%", echo = FALSE, fig.alt= "Reproducible code is: Readable meaning it is well-documented, uses well-known functions when appropriate, and follows a code style. Efficient meaning it doesn’t use up more computational resources than necessary and follows the advice of DRY (don’t repeat yourself). Consistent meaning it follows conventions and is organized"}
+ottrpal::include_slide("https://docs.google.com/presentation/d/1MNHf8JpolaEP_vQ_kB-1xRBF9wo3haCArRu117hBoHA/edit#slide=id.g221fb1e0594_0_6")
+```
+
+## Readable
+
+Reproducible code is readable code. In order for other people to use your code, they will need to be able to understand it. Because of that, code being readable is more important than code being innovative or clever.
+
+```{r, fig.align='center', out.width="100%", echo = FALSE, fig.alt= "Reproducible parrot is looking at their computer and says ‘It’s so easy for me to understand what is going on with this code because it is so readable and follows a style!’"}
+ottrpal::include_slide("https://docs.google.com/presentation/d/1MNHf8JpolaEP_vQ_kB-1xRBF9wo3haCArRu117hBoHA/edit#slide=id.g20ed8021e09_2_120")
+```
+
+### Well-documented
+
+Reproducible code is well documented code! This includes (but isn't limited to):
+
+- A README that can get individuals up to speed on the project quickly
+- Code and notebooks that have a healthy amount of comments
```

Comment:
maybe explain what notebooks are or remove it?

Reply:
I thought you were planning on adding a chapter before this about RMarkdowns and notebooks?

Reply:
We should definitely link that chapter when it exists.